core/shared/src/main/scala/zio/Scope.scala

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.collection.immutable.LongMap
+
+trait Scope extends Serializable { self =>
```

Comment:
I got distracted during your **awesome** talk so I might have missed it, but is it also going to replace `ZScope`?

Reply:
That is definitely something we are looking at. That goes to the point I made that potentially forking a fiber could return a scoped effect. This would potentially be a significant further simplification since it would subsume the existing semantics of `fork`, `forkdaemon`, `transplant`, and so on.

However, to prevent memory leaks it would require `Scope` supporting a notion of weak references. It would also result in creating a scoped effect every time you forked a fiber, which in a way would be very straighforward and elegant but we would also have to think through the semantics of.

Will be working on this week.